### PR TITLE
Update uglifier to version 3 (uglify-es)

### LINKED ISSUE #8157: FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
Hello Guys, 
I have update Meteor to 1.4.2.3 but on server show error : 

```
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory

<--- Last few GCs --->

  252470 ms: Mark-sweep 1340.5 (1458.0) -> 1339.6 (1458.0) MB, 566.4 / 0 ms [allocation failure] [scavenge might not succeed].
  253033 ms: Mark-sweep 1339.6 (1458.0) -> 1340.4 (1458.0) MB, 562.2 / 0 ms [allocation failure] [scavenge might not succeed].
  253612 ms: Mark-sweep 1340.4 (1458.0) -> 1337.4 (1458.0) MB, 579.0 / 0 ms [last resort gc].
  254190 ms: Mark-sweep 1337.4 (1458.0) -> 1339.1 (1458.0) MB, 578.4 / 0 ms [last resort gc].


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x2628e1eb4629 <JS Object>
    1: slowToString [buffer.js:413] [pc=0x270af61dbc15] (this=0x1dfd38efdf69 <an Uint8Array with map 0x2f5cd779741>,encoding=0x2628e1ebd761 <String[4]: utf8>,start=3777,end=401427)
    2: toString [buffer.js:~440] [pc=0x270af6595805] (this=0x1dfd38efdf69 <an Uint8Array with map 0x2f5cd779741>)
    3: arguments adaptor frame: 3->0
    4: /* anonymous */(aka /* anonymous */) [/bundle/bundle/progr...
```

Any sugesstion to fixed this issue ?
Thanks anyway

### PR DESCRIPTION
**Pull request by sebakerckhof**
Babili seems to use ridiculous amounts of memory: https://github.com/meteor/meteor/issues/8157
+I also hit some bugs with babili minification.

I experimented with https://github.com/Rich-Harris/butternut (`seba:butternut-minifier-js-plugin@2.0.0`) . A new minifier claiming to be es6+ compatible and 3x faster than Uglify, 10x faster than babili.
However, besides it being very new, I also got into memory issues.

UglifyJS3 (also named `uglify-es`) seems to use the least amount of memory and never gave an error minifying my ES6 code. Therefore, I propose to replace `uglify-js` with `uglify-es`.

I left babili in there since `uglify-es` is still quite new and `babili` might possibly catch and solve some `uglify-es` errors. However, once `uglify-es` is proven to be stable enough, we probably want to throw `babili` out (except if we want to minify es2016+)

Closes #8157.

### CODE REVIEW

**Comment by benjamn** [comment]
I would support this if `uglify-es` is ready (finally 🎉 ), though I think it should wait until 1.5.1.

**Comment by abernix** [comment]
I've been following this via https://github.com/mishoo/UglifyJS2/issues/448 for quite some time now.  It was just officially renamed this past week, I believe.

I've inched my `abernix:standard-minifier-js` along the `harmony` branch and while I haven't received any disconcerting updates from anyone using my minifier (over `standard-minifier-js`), I don't understand how UglifyJS is any more complete than it was 6 months ago when it showed deficiencies:

* [Tests from one month ago](https://gist.github.com/avdg/c72693832948e4725acea58fa6d10342)
* [Tests from six months ago](https://gist.github.com/avdg/2fa093526c610c2c63c49122f1c9ef57)

Both of those test suites are run against [`test262`](https://github.com/tc39/test262) and according to the updates, UglifyJS has a higher percentage of failures than it did six months ago.

That said, Babili is an absolute memory sink on larger files and `butternut` is too new.

File: packages/minifier-js/package.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ Package.describe({
 });
 
 Npm.depends({
-  "uglify-js": "3.0.13"
+  "uglify-es": "3.0.13"
```
Comment (benjamn) [inline packages/minifier-js/package.js]: Thanks!

**Comment by cesargalindo** [comment]
I've encountered a minification issue with the npm socket.io-client module.  Has anyone else has issues with the npm module?    https://www.npmjs.com/package/socket.io-client

Just adding socket.io-client to any Meteor Angular2/4 "Typescript" application breaks the production minification: 
import * as io from 'socket.io-client';
           or
import io from 'socket.io-client';
            or
let io = require('socket.io-client');

Note, the build is successful; however a blank page is displayed on the browser and the error "Uncaught ReferenceError: Z is not defined at delay.js" is displayed in the console.

The app within the browser after launching "node main.js"

**Comment by abernix** [comment]
@cesargalindo Please try `abernix:standard-minifier-js` as a replacement for `standard-minifier-js` and see if it resolves the problem you're experiencing.

As of right now, in Meteor 1.6 we're planning on officially switching to `uglify-es` (which `abernix:standard-minifier-js` implements via the same changes staged in this PR) so your feedback would be much appreciated!

**Comment by cesargalindo** [comment]
Awesome sauce @abernix ,  I replaced `standard-minifier-js `as suggested and all my apps build successfully now!  THANKS!

```
  >  meteor remove standard-minifier-js 
  >  meteor add abernix:standard-minifier-js
```

**Comment by fatso83** [comment]
Upgrading to Meteor 1.5 caused several production build problems, as the build broke due to:
1. maxing memory of Node process (fixed by env var increasing memory to 4GB)
2. [Node 4.8 incompatible with `node-gyp`](https://github.com/EyalAr/lwip/issues/297#issuecomment-300206001) (fixed by using pure js lib)
3. Minification errors resulting in issues similar to #8537 and #8536 
4. Almost triple build times (not breaking, but still 💩)

I just wanted to report that @cesargalindo's [steps to use the new standard minification](https://github.com/meteor/meteor/pull/8698#issuecomment-315213059) fixed the minification problem. Hope this makes it to a Meteor 1.5.2 patch release!

**Comment by davidworkman9** [comment]
I'm having serious issues with build times. I swapped out standard-minifiers for abernix's but did not notice much of an improvement, if any at all.

My build server is running on a t2.micro, use to take around 8-9 minutes. After upgrading from 1.4.2.3 to 1.5.1 on that same micro instance builds now take over 4.5 hours. I thought at first that I had decent build times on 1.5 but I don't think I did. I switched to abernix's this morning and ran another build but stopped it after 2 hours had elapsed as I needed to get it pushed out.

Occasionally if I needed to push something out right away I'd switch my micro to a c4.large and it'd run a build in around 2 minutes. Ran one yesterday on the standard minifier package, took around 8.5 minutes. Ran another one this morning with abernix's and it took 7.5 minutes. A bit better but still a far cry from my 2 minute builds I was experiencing before.

**Comment by abernix** [comment]
@sebakerckhof Do you mind `git pull`-ing and `rebase`-ing this off the latest `release-1.6` branch? 😃

**Comment by abernix** [comment]
Actually, nevermind.  I'll just land it in `devel`.  :smile:

**Pull request merged**